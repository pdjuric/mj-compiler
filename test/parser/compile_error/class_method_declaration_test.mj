program main_test

class A {
    int fieldA;
    {
        void foo(int a, int b) {}
        int bar() int t, a, b, s; {}
        A baz(A a) {}
    }
}

class X extends A {
    {


        int bar(int visakParam) {}
        A bar() {}

        A baz() A a; {}
        A baz(X a) {}
        X baz(A a) {}                   // ok

        A qux() {}                      // ne overrideuje ni jedan metod - ok
    }
}

class XX extends X {
    {
        A qux() {}
    }
}


class Y extends A{
    int fieldY;
    {
        void Name() {}
        void fieldA() {}
        int fieldY() {}

        int f() char f; {}
    }
}


{
    void main() {}
}