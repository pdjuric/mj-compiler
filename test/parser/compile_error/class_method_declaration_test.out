ERROR 15:39:12,129 - line  16 - Excess parameter visakParam in override function bar.
ERROR 15:39:12,130 - line  17 - Override of function bar in class X has incompatible return type.
ERROR 15:39:12,130 - line   0 - OverrideParametersMissing  baz
ERROR 15:39:12,130 - line  20 - OverrideParameterTypeMismatch a baz